api/oneclick.py

```diff
@@ -0,0 +1,707 @@
+                else:
+                    raise RemoteInitiatedServerError(message, action)
+
+            # a 409 conflict code can mean many things
+            if response.status_code == 409 and action == 'checkout':
+                if message == "Maximum checkout count reached.":
+                    raise PatronLoanLimitReached(action + ": " + message)
+                elif message == "Checkout item already exists":
+                    # we tried to borrow something the patron already has
+                    raise AlreadyCheckedOut(action + ": " + message)
+                else:
+                    raise CannotLoan(action + ": " + message)
+
+            if response.status_code == 409 and action == 'checkin':
+                if message == "Checkout does not exists or it is already terminated or expired.":
+                    # we tried to return something the patron doesn't own
+                    raise NotCheckedOut(action + ": " + message)
+                else:
+                    raise CannotReturn(action + ": " + message)
+
+            if response.status_code == 404:
+                raise NotFoundOnRemote(action + ": " + message)
+
+            if response.status_code == 400:
+                raise InvalidInputException(action + ": " + message)
+
+        elif message:
+            # http code was OK, but info wasn't sucessfully read from db
+            if message.startswith("eXtensible Framework was unable to locate the resource for RB.API.OneClick.UserPatron.Get"):
+                raise PatronNotFoundOnRemote(action + ": " + message)
+            else:
+                self.log.warning("%s not retrieved: %s ", action, message)
+                raise CirculationException(action + ": " + message)
+
+
+    def queue_response(self, status_code, headers={}, content=None):
+        """ Allows smoother faster creation of unit tests by letting 
+        us live-test as we write. """
+        pass
+
+
+
+class MockOneClickAPI(BaseMockOneClickAPI, OneClickAPI):
+    pass
+
+
+
+class OneClickCirculationMonitor(Monitor):
+    """Maintain LicensePools for OneClick titles.
+
+    Bibliographic data isn't inserted into new LicensePools until
+    we hear from the metadata wrangler.
+    """
+    def __init__(self, _db, name="OneClick Circulation Monitor",
+                 interval_seconds=500,
+                 maximum_consecutive_unchanged_books=None):
+        super(OneClickCirculationMonitor, self).__init__(
+            _db, name, interval_seconds=interval_seconds)
+        self.maximum_consecutive_unchanged_books = (
+            maximum_consecutive_unchanged_books)
+
+
+    def recently_changed_ids(self, start, cutoff):
+        return self.api.get_delta(start, cutoff)
+
+
+    def run(self):
+        self.api = OneClickAPI(self._db)
+        super(OneClickCirculationMonitor, self).run()
+
+    def run_once(self, start, cutoff):
+        pass
```

Comment:
monitor is started in the oneclick_monitor brach, pull request https://github.com/NYPL-Simplified/circulation/pull/421/files